pkg/tests/tasks/migration/migration_test.go

```diff
@@ -90,45 +114,158 @@ spec:
 		oc.DefaultOC.WaitFor(t, smcp.Namespace, "ServiceMeshMemberRoll", "default", `jsonpath='{.status.configuredMembers[?(@=="bookinfo")]}'`)
 
 		t.LogStep("Create 3.0 controlplane and IstioCNI")
-		istioTempl := `apiVersion: sailoperator.io/v1alpha1
-kind: Istio
+		setupIstio(t, istio)
+
+		t.LogStep("Install bookinfo and bookinfo gateway")
+		app.InstallAndWaitReady(t, app.Bookinfo(ns.Bookinfo))
+
+		// update default gateway
+		oc.ApplyFile(t, ns.Bookinfo, migrationGateway)
+		oc.DefaultOC.WaitDeploymentRolloutComplete(t, ns.Bookinfo, "bookinfo-gateway")
+		oc.DefaultOC.WaitFor(t, ns.Bookinfo, "Route", "bookinfo-gateway", `jsonpath="{.status.ingress[].host}"`)
+		hostname := oc.GetJson(t, ns.Bookinfo, "Routes", "bookinfo-gateway", "{.spec.host}")
+		bookinfoGatewayURL := fmt.Sprintf("http://%s/productpage", hostname)
+		continuallyRequest(t, bookinfoGatewayURL)
+
+		t.LogStep("Migrate bookinfo to 3.0 controlplane")
+		ossm3RevName := oc.GetJson(t, "", "Istio", istio.Name, "{.status.activeRevisionName}")
+		oc.Label(t, "", "Namespace", ns.Bookinfo, "istio-injection- istio.io/rev="+ossm3RevName)
+		// Wait for book info to be removed.
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			var members []string
+			output := oc.GetJson(t, smcp.Namespace, "ServiceMeshMemberRoll", "default", "{.status.configuredMembers}")
+			if err := json.Unmarshal([]byte(output), &members); err != nil {
+				t.Error(err)
+			}
+			contains := false
+			for _, member := range members {
+				if member == "bookinfo" {
+					contains = true
+					break
+				}
+			}
+			if contains {
+				t.Error("bookinfo found in SMMR. Expected it to be removed.")
+			}
+		})
+
+		workloads := []workload{
+			{Name: "productpage-v1", Labels: map[string]string{"app": "productpage", "version": "v1"}},
+			{Name: "reviews-v1", Labels: map[string]string{"app": "reviews", "version": "v1"}},
+			{Name: "reviews-v2", Labels: map[string]string{"app": "reviews", "version": "v2"}},
+			{Name: "reviews-v3", Labels: map[string]string{"app": "reviews", "version": "v3"}},
+			{Name: "ratings-v1", Labels: map[string]string{"app": "ratings", "version": "v1"}},
+			{Name: "details-v1", Labels: map[string]string{"app": "details", "version": "v1"}},
+			{Name: "bookinfo-gateway", Labels: map[string]string{"istio": "bookinfo-gateway"}},
+		}
+		oc.DefaultOC.RestartDeployments(t, ns.Bookinfo, workloadNames(workloads)...)
+		// Waiting for the rollouts to complete ensures that old pods have been deleted.
+		// If there are old pods lying around then the assertion below to get the pod annotations
+		// will fail.
+		oc.WaitDeploymentRolloutComplete(t, ns.Bookinfo, workloadNames(workloads)...)
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			if output := oc.DefaultOC.Invokef(t, `kubectl get pods -n %s -o jsonpath='{.items[?(@.metadata.deletionTimestamp!="")].metadata.name}'`, ns.Bookinfo); output != "" {
```

Comment:
We will run this only against OCP so maybe `oc` should be used instead of `kubectl` everywhere.